### PR TITLE
[misc] [refactor] Replace the word tensor by field in comments and error messages

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #1500

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
This PR only focus on comments and error messages.
The term `tensor` in codes are non-trivial and will be put in a separate PR:
```
taichi/ir/frontend_ir.h:class ExternalTensorExpression : public Expression {
taichi/ir/frontend_ir.h:  ExternalTensorExpression(const DataType &dt, int dim, int arg_id)
taichi/ir/expr.cpp:  TI_ASSERT(is<GlobalVariableExpression>() || is<ExternalTensorExpression>());
taichi/ir/ir.cpp:    TI_ASSERT(var.is<ExternalTensorExpression>());
taichi/ir/ir.cpp:        var.cast<ExternalTensorExpression>()->stmt, index_stmts));
taichi/backends/opengl/opengl_api.cpp:                     taichi::starts_with(kernel_name, "tensor_to_") ||
taichi/backends/cuda/codegen_cuda.cpp:            // Note: both numpy and PyTorch support arrays/tensors with zeros
taichi/python/export_lang.cpp:           [](Expr *expr) { return expr->is<ExternalTensorExpression>(); })
taichi/python/export_lang.cpp:  m.def("make_external_tensor_expr",
taichi/python/export_lang.cpp:        Expr::make<ExternalTensorExpression, const DataType &, int, int>);
tests/python/test_numpy_io.py:    # Taichi tensors
tests/python/test_numpy_io.py:    # For PyTorch tensors, use to_torch/from_torch instead
tests/python/test_torch_ad.py:        X = torch.tensor(2 * np.ones((n, ), dtype=np.float32),
tests/python/test_torch_ad.py:        X = torch.tensor(2 * np.ones((n, ), dtype=np.float32),
tests/python/test_torch_io.py:        y = torch.Tensor(np.ones(shape=n, dtype=np.int32)).to(device)
tests/python/test_torch_io.py:            inp, = ctx.saved_tensors
tests/python/test_torch_io.py:    X = torch.tensor(2 * np.ones((n, ), dtype=np.float32), requires_grad=True)
tests/python/test_torch_io.py:    X = torch.tensor(2 * np.ones((n, n), dtype=np.float32), requires_grad=True)
tests/python/test_torch_io.py:    X = torch.tensor(2 * np.ones((n, n, n), dtype=np.float32),
tests/python/test_torch_io.py:    t1 = torch.tensor(2 * np.ones((n, n), dtype=np.float32))
tests/python/test_torch_io.py:    t2 = torch.tensor(2 * np.ones((n, n, 2, 3), dtype=np.float32))
misc/scripts/cnn_compare/scn.py:input_voxel = torch.tensor(input_voxel)
docs/scalar_tensor.rst:.. _scalar_tensor:
docs/scalar_tensor.rst:   We once used the term **tensor** instead of **field**. **Tensor** will no longer be used.
docs/tensor_matrix.rst:.. _tensor:
docs/tensor_matrix.rst:* See :ref:`scalar_tensor` for more details.
docs/compilation.rst:  Examples include ``x.to_numpy()`` and ``y.from_torch(torch_tensor)``. When you invoke these functions,
docs/vector.rst:In fact, ``Vector`` is simply an alias of ``Matrix``, just with ``m = 1``. See :ref:`matrix` and :ref:`tensor` for more details.
docs/vector.rst:    :parameter shape: (optional, scalar or tuple) shape of the vector field, see :ref:`tensor`
docs/vector.rst:    In Python-scope, ``ti.field`` declares a scalar field (:ref:`scalar_tensor`), while ``ti.Vector.field`` declares a vector field.
docs/external.rst:External arrays refers to NumPy arrays or PyTorch tensors.
docs/external.rst:We also provide interfaces to copy data directly between Taichi fields and PyTorch tensors.
docs/external.rst:   :parameter device: (torch.device) the device where the PyTorch tensor is stored
docs/external.rst:   :return: (torch.Tensor) The PyTorch tensor containing data in ``x``.
docs/external.rst:.. function:: field.from_torch(tensor)
docs/external.rst:   :parameter tensor: (torch.Tensor) The PyTorch tensor with data to initialize the field
docs/external.rst:Shapes of Taichi fields (see :ref:`scalar_tensor`) and those of corresponding NumPy arrays are closely connected via the following rules:
docs/differentiable_programming.rst:Unlike tools such as TensorFlow where **immutable** output buffers are generated, the **imperative** programming paradigm adopted in Taichi allows programmers to freely modify global fields.
docs/layout.rst:Fields (:ref:`scalar_tensor`) can be *placed* in a specific shape and *layout*.
docs/install.rst:  You might be using Ubuntu 16.04, please try the solution in `this thread <https://github.com/tensorflow/serving/issues/819#issuecomment-377776784>`_:
docs/meta.rst:See :ref:`scalar_tensor` for more details.
docs/hello.rst:See :ref:`scalar_tensor` for more details on fields.
docs/matrix.rst:    :parameter shape: (optional, scalar or tuple) shape of the matrix field, see :ref:`tensor`
docs/matrix.rst:    In Python-scope, ``ti.field`` declares a scalar field (:ref:`scalar_tensor`), while ``ti.Matrix.field`` declares a matrix field.
docs/index.rst:   tensor_matrix
docs/index.rst:   scalar_tensor
python/taichi/torch_io.py:def from_torch_template(expr: ti.template(), torch_tensor: ti.ext_arr()):
python/taichi/torch_io.py:        expr[i] = torch_tensor[i]
python/taichi/torch_io.py:def to_torch_template(expr: ti.template(), torch_tensor: ti.ext_arr()):
python/taichi/torch_io.py:        torch_tensor[i] = expr[i]
python/taichi/torch_io.py:def from_torch(expr, torch_tensor):
python/taichi/torch_io.py:    expr.from_torch_(torch_tensor)
python/taichi/torch_io.py:def to_torch(expr, torch_tensor):
python/taichi/torch_io.py:    expr.to_torch_(torch_tensor)
python/taichi/lang/expr.py:    @deprecated('x.get_tensor_members()', 'x.get_field_members()')
python/taichi/lang/expr.py:    def get_tensor_members(self):
python/taichi/lang/expr.py:        from .meta import fill_tensor
python/taichi/lang/expr.py:        fill_tensor(self, val)
python/taichi/lang/expr.py:    #@deprecated('tensor.parent()', 'tensor.snode.parent()')
python/taichi/lang/expr.py:        from .meta import tensor_to_ext_arr
python/taichi/lang/expr.py:        tensor_to_ext_arr(self, arr)
python/taichi/lang/expr.py:        from .meta import tensor_to_ext_arr
python/taichi/lang/expr.py:        tensor_to_ext_arr(self, arr)
python/taichi/lang/expr.py:        from .meta import ext_arr_to_tensor
python/taichi/lang/expr.py:        ext_arr_to_tensor(arr, self)
python/taichi/lang/expr.py:        from .meta import tensor_to_tensor
python/taichi/lang/expr.py:        tensor_to_tensor(self, other)
python/taichi/lang/matrix.py:                ' See https://taichi.readthedocs.io/en/stable/tensor_matrix.html#matrix-size'
python/taichi/lang/matrix.py:    @deprecated('x.get_tensor_members()', 'x.get_field_members()')
python/taichi/lang/matrix.py:    def get_tensor_members(self):
python/taichi/lang/matrix.py:            """Proxy when a tensor of Matrices is accessed by host."""
python/taichi/lang/matrix.py:        # Took `self.entries[0]` as a representation of this tensor-of-matrices.
python/taichi/lang/matrix.py:    def from_torch(self, torch_tensor):
python/taichi/lang/matrix.py:        return self.from_numpy(torch_tensor.contiguous())
python/taichi/lang/matrix.py:        from .meta import tensor_to_tensor
python/taichi/lang/matrix.py:        tensor_to_tensor(self, other)
python/taichi/lang/kernel.py:                        assert has_torch and isinstance(v, torch.Tensor)
python/taichi/lang/kernel.py:                            # External tensor on cuda
python/taichi/lang/kernel.py:                            # External tensor on cpu
python/taichi/lang/kernel.py:            has_array = isinstance(v, torch.Tensor)
python/taichi/lang/kernel_arguments.py:    def __init__(self, tensor=None, dim=None):
python/taichi/lang/kernel_arguments.py:        self.tensor = tensor
python/taichi/lang/kernel_arguments.py:    return Expr(taichi_lang_core.make_external_tensor_expr(dtype, dim, id))
python/taichi/lang/meta.py:def fill_tensor(tensor: ti.template(), val: ti.template()):
python/taichi/lang/meta.py:    for I in ti.grouped(tensor):
python/taichi/lang/meta.py:        tensor[I] = val
python/taichi/lang/meta.py:def tensor_to_ext_arr(tensor: ti.template(), arr: ti.ext_arr()):
python/taichi/lang/meta.py:    for I in ti.grouped(tensor):
python/taichi/lang/meta.py:        arr[I] = tensor[I]
python/taichi/lang/meta.py:def tensor_to_image(tensor: ti.template(), arr: ti.ext_arr()):
python/taichi/lang/meta.py:    for I in ti.grouped(tensor):
python/taichi/lang/meta.py:        t = cook_image_type(tensor[I])
python/taichi/lang/meta.py:def tensor_to_tensor(tensor: ti.template(), other: ti.template()):
python/taichi/lang/meta.py:    for I in ti.grouped(tensor):
python/taichi/lang/meta.py:        tensor[I] = other[I]
python/taichi/lang/meta.py:def ext_arr_to_tensor(arr: ti.ext_arr(), tensor: ti.template()):
python/taichi/lang/meta.py:    for I in ti.grouped(tensor):
python/taichi/lang/meta.py:        tensor[I] = arr[I]
python/taichi/misc/gui.py:                from taichi.lang.meta import tensor_to_image
python/taichi/misc/gui.py:                tensor_to_image(img, self.img)
```

### CODE REVIEW

File: taichi/ir/expr.cpp [hunk 1]
```diff
@@ -183,7 +183,7 @@ Expr ptr_if_global(const Expr &var) {
     // singleton global variable
     TI_ASSERT_INFO(var.snode()->num_active_indices == 0,
                    "Please always use 'x[None]' (instead of simply 'x') to "
-                   "access any 0-D tensor.");
+                   "access any 0-D fields.");
```
Comment (yuanming-hu) [inline taichi/ir/expr.cpp:186]:
```suggestion
                   "access any 0-D field.");
```

File: python/taichi/tools/np2ply.py [hunk 2]
```diff
@@ -1,278 +1,278 @@
...
+        real_path = path + "_" + "{0:0=6d}".format(series_num) + ".ply"
+        self.export_ascii(real_path)
+
+    def export_frame(self, series_num: int, path: str):
+        # if path has ply ending
+        last_4_char = path[-4:]
+        if last_4_char == ".ply":
+            path = path[:-4]
+
+        real_path = path + "_" + "{0:0=6d}".format(series_num) + ".ply"
```
Comment (Rullec) [inline python/taichi/tools/np2ply.py:277]: Oh, what makes the git have such a big changeset? Did we do something unexpected? ;O

**Comment by Rullec** [comment]
Nice work, LGTM!

File: python/taichi/tools/np2ply.py (hunk 2 above)
Comment (archibate) [reply python/taichi/tools/np2ply.py:277]: Our format server @taichi-gardener is replacing CRLF by LF introduced in #1706. Let's revert this part and let the code format to fail for now to pvc #1728.

**Pull request merged**